resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/service/TemporaryResourceStorageService.kt

```diff
@@ -62,14 +65,31 @@ class TemporaryResourceStorageService {
     }
 
     fun getResourceMetadata(id: String): Map<String, Any> {
-        val metaDataFile = Path(id)
+        val metaDataFile = getMetaDataFileFromResourceId(id)
         if (metaDataFile.notExists()) {
             throw IllegalArgumentException("No resource found with id '$id'")
         }
         val typeRef = object : TypeReference<Map<String, Any>>() {}
         return Mapper.INSTANCE.get().readValue(metaDataFile.readText(), typeRef)
     }
 
+    internal fun getMetaDataFileFromResourceId(resourceId: String): Path {
+        val resourceUuid = UUID.fromString(resourceId)
+        if (resourceUuid.leastSignificantBits != resourceUuid.mostSignificantBits) {
+            throw IllegalArgumentException("Invalid resourceId: $resourceId")
+        }
+        val fileNumber = resourceUuid.leastSignificantBits.toULong()
+        return Path.of(TEMP_DIR.pathString, "temporaryResourceMetadata$fileNumber.json")
+    }
+
+    private fun getResourceIdFromMetaDataFile(tempFile: Path): String {
+        if (!tempFile.nameWithoutExtension.startsWith("temporaryResourceMetadata")) {
+            throw IllegalArgumentException("File is not a temporaryResourceMetadata file: $tempFile")
+        }
+        val fileNumber = tempFile.nameWithoutExtension.substring("temporaryResourceMetadata".length).toULong().toLong()
```

Comment:
I'd prefer to have this not be a UUID based on the randomly generated name. The Java documentation says
```
The details as to how the name of the file is constructed is implementation dependent and therefore not specified. Where possible the prefix and suffix are used to construct candidate names in the same manner as the File.createTempFile(String, String, File) method.
```
So i think we can't 100% expect this to be a long. I think we should just return the file name (not the path).

Reply:
Changed